ktor-server/ktor-server-plugins/ktor-server-double-receive/jvmAndNix/src/io/ktor/server/plugins/doublereceive/DoubleReceiveConfig.kt

```diff
@@ -24,6 +24,13 @@ public class DoubleReceiveConfig {
      */
     public var cacheRawRequest: Boolean = true
 
+    /**
+     * Indicates if a file should be used to cache raw requests. Otherwise, requests will be stored in memory.
+     *
+     * If [cacheRawRequest] is `false` then this setting is ignored.
+     */
+    public var useFileForCache: Boolean = PlatformUtils.IS_JVM
```

Comment:
isn't it better to make it configurable per request, and not via one boolean flag?
similar as with `excludeFromCache`?
So `File` cache will be used only for those requests, that can contain big body. If not, using file cache will decrease performance, as writing to File is almost always slower than writing to Memory

Reply:
You can install the plugin only for the specific route which is supposed to have a big body. I'm not sure how it should be possible to configure for a specific request.

Reply:
i mean use the same declaration as with `excludeFromCache`, so like:
```kotlin
fun useFileForCache(block: (call: ApplicationCall, body: Any) -> Boolean)
```
and in this case we will be able to f.e. do `file` cache when `contentLength` is big, or for specific content type, and not to apply this config for multiple routes

Reply:
I agree, it should be configurable per request. Also, default should be opposite, IMO.

Reply:
@olme04  
`fun useFileForCache(block: (call: ApplicationCall, body: Any) -> Boolean)` is actually a good idea. Thank you for the hint.

@rsinukov the file default seems to be not much slower, but much safer. I would keep it as a default, but we can discuss.

Reply:
Yep. And I would even put default here:`fun useFileForCache(block: (call: ApplicationCall) -> Boolean = { true })`